client/relay.go

```diff
@@ -0,0 +1,58 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/pokt-network/shannon-sdk/sdk"
+)
+
+var _ sdk.RelayClient = (*relayClient)(nil)
+
+// relayClient is a RelayClient implementation that uses the HTTP client
+// of the Go standard library.
+type relayClient struct {
+	httpClient *http.Client
+}
+
+// NewRelayClient creates a new relay client with the default HTTP client, which
+// is used to send the relay requests.
+func NewRelayClient() (sdk.RelayClient, error) {
```

Comment:
The ability to supply a custom HTTP client is required for the grove gateway so we can control timeouts.

Reply:
_tl;dr Happy to see what a change would look like but let's evaluate first._

Thanks for the comment @ezeike.

I want to call out that we are building (and designing) a general purpose SDK for all gateways (portal and other) to integrate with. Our goal is to 

With that said, few questions:

1. Are you 100% sure that they right call for all future gateways is to provide their own HTTP client?
2. Regardless of the answer to (1), can you think of a way to provide it as an option?
3. Regardless of the answers to (1) and (2), Is there a way whereby portal would modify it's implementation to accommodate this interface.

Reply:
Of course, this is just a simplistic implementation of the `sdk.RelayClient`, any serious gateway should provide its own.

I should maybe rename this directory to `example-clients/` instead of `client/`

Reply:
1.  Yes 100% sure the ability to supply your own client is the right call for all future gateways. Required, no, but I wouldn't object to that either.  Everyone will change the default connection timeout of 30s and we've also had to make changes to the write timeout.

I think it's worth a small effort here in order to avoid a default-unusable-in-prod component in the SDK.

2. The Shannon light client currently in the gateway uses the functional options pattern for this. I'd lean towards that approach.

3. Sure we can implement our own relay client.